plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/transactionlog/ProtocolEntry.java

```diff
@@ -61,4 +73,73 @@ public boolean writerFeaturesContains(String featureName)
     {
         return writerFeatures.map(features -> features.contains(featureName)).orElse(false);
     }
+
+    public static Builder builder(ProtocolEntry protocolEntry)
+    {
+        return new Builder(protocolEntry.minReaderVersion, protocolEntry.minWriterVersion, protocolEntry.readerFeatures, protocolEntry.writerFeatures);
+    }
+
+    public static Builder builder(int readerVersion, int writerVersion)
+    {
+        return new Builder(readerVersion, writerVersion, Optional.empty(), Optional.empty());
+    }
+
+    public static class Builder
+    {
+        private int readerVersion;
+        private int writerVersion;
+        private final Set<String> readerFeatures = new HashSet<>();
+        private final Set<String> writerFeatures = new HashSet<>();
+
+        public Builder(int readerVersion, int writerVersion, Optional<Set<String>> readerFeatures, Optional<Set<String>> writerFeatures)
+        {
+            this.readerVersion = readerVersion;
+            this.writerVersion = writerVersion;
+            requireNonNull(readerFeatures, "readerFeatures is null").ifPresent(this.readerFeatures::addAll);
```

Comment:
It adds feature without checking if `readerVersion` and `writerVersion` as in separate `enable....` Should caller of Builder assure that those version are consistent with features?
Or maybe this constructor should be made private?